src/PlaylistSave.hxx

```diff
@@ -31,32 +31,38 @@ playlist_print_song(BufferedOutputStream &os, const DetachedSong &song);
 void
 playlist_print_uri(BufferedOutputStream &os, const char *uri);
 
-/**
- * Saves a queue object into a stored playlist file.
- */
-void
-spl_save_queue(const char *name_utf8, const Queue &queue);
+struct PlaylistSaveMode {
+private:
+	bool create = false;
+	bool replace = false;
+	bool append = false;
 
-/**
- * Saves a playlist object into a stored playlist file.
- */
-void
-spl_save_playlist(const char *name_utf8, const playlist &playlist);
+	const char *argValidationError = "Expected one of 'create', 'append', 'replace'";
+
+public:
+	[[nodiscard]] static PlaylistSaveMode parseArg(const char* arg);
+
+	[[nodiscard]] inline bool is_valid() const noexcept {
+		return create || (append != replace);
+	}
+
+	[[nodiscard]] inline bool is_create_only() const noexcept {
+		return create && !(append || replace);
+	}
+
+	[[nodiscard]] inline bool is_existing_only() const noexcept {
+		return !create && (append || replace);
+	}
+
+	[[nodiscard]] inline bool is_append() const noexcept {
+		return append;
+	}
+};
 
-/**
- * Store a queue object into a stored playlist file.
- *
- * @arg replace true to relace the stored playlist contents, false to append to the list.
- */
 void
-spl_store_queue(const char *name_utf8, bool replace, const Queue &queue);
+spl_save_queue(const char *name_utf8, const PlaylistSaveMode& save_mode, const Queue &queue);
```

Comment:
Pass this by value, not by reference - hey, even if you keep this as struct with 3 bools, it's just 3 bytes (1 bytes for an `enum : uint8_t`) - the reference will take 8 bytes for the pointer, plus the size of the struct!